amnahhashmi/shield
# Errand confirmation

To Do:
- [x] Connect celery task to Twilio
- [x] Create handler to receive YES/NO from requestor
- [x] Record in DB that errand has been confirmed complete by requestor
- [x] Ensure requestor flow isn't initiated for an already active flow
- [x] 🏁 Clean back up settings file
- [x] 🏁 Remove CSRF exemption from API call

Dependencies:
* ☎️ ~~Req_Happy_VolDelivered flow needs to be published in Twilio~~
* ☎️ ~~Phone number needed in Twilio~~

Test Plan:
1. Ensure there is a "live" errand in your DB instance. Essentially where `requestor_confirmed` field is NULL and `status` equals 3.
2. Ensure the requestor record tied to the errand has a phone number you have access to check SMS messages.
3. `celery -A shield worker --pool=solo -B -l info`. A task should be fired off which executes a flow in Twilio to send you a message.
4. You should be texted an SMS reading:
    "[VOLUNTEER NAME] has completed your delivery. So that we know you’re OK, reply with the # (e.g. “1”) that matches your current status.
    1 = Yes, I received my delivery!
    0 = No, I’m still waiting
    You can also reach us here if you have any questions"
    where VOLUNTEER_NAME is populated.
5. The errand record in the DB from step 1 should now have `requestor_confirmed` as `False`
6. Respond 1 to the text message.
7. Repeat step 5, but the value of `requestor_confirmed` should now be `True`
